torchao/quantization/quant_primitives.py

```diff
@@ -130,17 +108,32 @@ def _get_reduction_params(block_size, input_size):
             cur_dim += 1
     return shape_for_reduction, reduction_dims
 
+def register_custom_op(name: str):
+    from torch._inductor.decomposition import register_decomposition
+
+    def decorator(fn):
+        if TORCH_VERSION_AFTER_2_5:
+            opdef = torch.library.custom_op(name, mutates_args=())(fn)
+            opdef.register_fake(fn)
+            register_decomposition([opdef._opoverload])(fn)
```

Comment:
@supriyar here we register a decomp for inductor, so in torch.compile these ops will still be decomposed